Dockerfile

```diff
@@ -0,0 +1,25 @@
+# syntax=docker/dockerfile:1
+FROM --platform=$BUILDPLATFORM golang:1.17 as build
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+
+WORKDIR /go/src/dnsproxy/
+
+COPY . .
+
+RUN set -e \
+    && echo "Running on $BUILDPLATFORM, building for $TARGETPLATFORM" \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y ruby \
+    && ruby docker-env.rb
+
+FROM --platform=$TARGETPLATFORM busybox:1.35
+COPY --from=build /go/src/dnsproxy/dnsproxy /usr/local/bin/dnsproxy
+
+WORKDIR /root
+
+EXPOSE 53/udp
+EXPOSE 443
+EXPOSE 853
+
```

Comment:
It's better to set the binary as the entrypoint here with an explicit path.

This lets the user specify a path to a configuration file.

```Dockerfile
ENTRYPOINT ["/usr/local/bin/dnsproxy"]
```